src/modules/module_utils.c

```diff
@@ -413,6 +422,50 @@ void module_strip_punctuation_default(char *buf)
 	module_strip_punctuation_some(buf, "~#$%^&*+=|<>[]_");
 }
 
+gchar *module_multicases_string_replace(
+	gchar *text,
+	const MulticasesString *mcstr)
+{
+	GRegex *regex;
+	GError *error = NULL;
+	gchar *result;
+
+	regex = g_regex_new(mcstr->pattern, G_REGEX_OPTIMIZE, 0, &error);
+	if (!regex) {
+		DBG("ERROR compiling regular expression: %s.", error->message);
+		g_error_free(error);
+		return text;
+	}
+
+	result = g_regex_replace(regex, text, -1, 0, mcstr->replace, G_REGEX_MATCH_DEFAULT, NULL);
```

Comment:
you want to look for errors. On error, just log it and return `text` as it is. Without error, you need to free `text`